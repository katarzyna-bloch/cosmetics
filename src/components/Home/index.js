import styled from 'styled-components'

const Photo = styled.img`
  position: absolute;
  left: 0;
  top: 0;
  max-height: 100%;
  width: 100%;
  object-fit: cover;
  /* max-width: 1024px; */
`

const Home = () => ( 
  <Photo src="photo.jpg" alt='kosmetyki' /> 
)

export default Home