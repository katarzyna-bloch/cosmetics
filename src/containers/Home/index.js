import styled from 'styled-components'
import React, { useEffect } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

import ContentHeader from '../../components/ContentHeader'
import Banners from '../../components/Banners'
import PageWrapper from '../../components/PageWrapper'
import ProductsSlider from '../../components/ProductsSlider'
import Preloader from '../../components/Preloader'
import { requestDashboardData } from '../../redux/actions'

const Heading = styled(ContentHeader)`
  font-size: 25px;
  margin: 20px 0;
`

export const Home = ({ fetchDashboardData, saleProducts, topProducts, banners, loading }) => {
  useEffect(() => {
    fetchDashboardData()
  }, [])
  
  if (loading) {
    return <Preloader />
  }

  return (
    <PageWrapper>
      <Banners banners={banners} />
      <Heading>W tym tygodniu w promocji</Heading>
      <ProductsSlider products={saleProducts} />
      <Heading>Top oferty</Heading>
      <ProductsSlider products={topProducts} />
    </PageWrapper>
  )
}

const mapStateToProps = ({ dashboard }) => ({
  ...dashboard,
})

const mapDispatchToProps = (dispatch) => ({
  fetchDashboardData: () => dispatch(requestDashboardData()),
})

Home.propTypes = {
  fetchDashboardData: PropTypes.func.isRequired,
  saleProducts: PropTypes.array.isRequired,
  topProducts: PropTypes.array.isRequired,
  banners: PropTypes.array.isRequired,
  loading: PropTypes.bool.isRequired,
}

export default connect(mapStateToProps, mapDispatchToProps)(Home)
