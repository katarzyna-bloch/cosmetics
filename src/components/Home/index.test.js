import { shallow } from "enzyme"
import toJson from "enzyme-to-json"
import Home from "."

it("should render correctly", () => {
  const wrapped = shallow(<Home />)
  expect(toJson(wrapped)).toMatchSnapshot()
})
