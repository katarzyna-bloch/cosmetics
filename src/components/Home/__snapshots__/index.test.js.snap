// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<styled.img
  alt="kosmetyki"
  src="photo.jpg"
/>
`;
